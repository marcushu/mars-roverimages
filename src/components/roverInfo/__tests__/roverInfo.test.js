import { render, screen, within} from "@testing-library/react";
import RoverInfo from "..";


it('should load the correct info card', () => {
    render(<RoverInfo selectedRover={'Curiosity'} />);

    const view = screen.getByText(  /in collaboration with the spirit rover \(launched jun 10, 2003\), opportunity set about covering over 28 miles or martian surface in search of evidence that water may have existed on the planet\. armed with a suite of geologists tools, including a rock abrasian tool and a microscopic camera for close analysis of surface samples, opportunity collected data from january 25, 2004 to june of 2018 before succumbing to a dust storm\./i  );
    const rovername = within(view).getByText(/curiosity/i); 

    expect(rovername).toBeInTheDocument();
});

it('should match snapshot', () => {
    const { asFragment } = render(<RoverInfo selectedRover={'Spirit'} />);

    expect(asFragment()).toMatchSnapshot();
})