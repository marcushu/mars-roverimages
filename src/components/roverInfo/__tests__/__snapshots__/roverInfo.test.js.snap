// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="container"
    >
      <div
        class="textbox"
      >
        <div
          class="rovertitle"
        >
          Spirit
        </div>
        As a counterpart to the opportunity rover, Spirit broadened the search for evidence of water which could have supported life on Mars. Spirit and Opportunity landed on opposite sides of the planet within 21 days of each other. Despite becoming stuck in soft sand, Spirit continued conducting research until march 22, 2010 after completing 2208 sols.
      </div>
      <div
        class="imagebox"
      >
        <img
          alt="rover pic"
          src="spirit.jpeg"
          style="border-radius: 20px; max-width: 450px;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
